executors/cpp/datetime_fmt.cpp

```diff
@@ -0,0 +1,257 @@
+    }
+
+    option_item = json_object_object_get(options_obj, "hour");
+    if (option_item) {
+      hour_str = json_object_get_string(option_item);
+      cout << "# hour: " << hour_str << endl;
+    }
+
+    option_item = json_object_object_get(options_obj, "minute");
+    if (option_item) {
+      minute_str = json_object_get_string(option_item);
+      cout << "# minute: " << minute_str << endl;
+    }
+
+    option_item = json_object_object_get(options_obj, "second");
+    if (option_item) {
+      second_str = json_object_get_string(option_item);
+      cout << "# second: " << second_str << endl;
+    }
+
+    option_item = json_object_object_get(options_obj, "fractional_seconds_digits");
+    if (option_item) {
+      fractional_seconds_digits = json_object_get_int(option_item);
+      cout << "# fractional seconds digits: " << fractional_seconds_digits << endl;
+    }
+
+  }
+
+  json_object *date_skeleton_obj = json_object_object_get(json_in, "datetime_skeleton");
+  if (date_skeleton_obj) {
+    // Data specifies a date time skeleton. Make a formatter based on this.
+    string skeleton_string = json_object_get_string(date_skeleton_obj);
+
+    cout << "# Skelecton = " << skeleton_string << endl;
+
+    UnicodeString u_skeleton = skeleton_string.c_str();
+    df = DateFormat::createInstanceForSkeleton(u_skeleton,
+                                               displayLocale,
+                                               status);
+    if (U_FAILURE(status)) {
+      // TODO: Return error.
+      cout << "# ERROR in createInstanceForSkeleton" << endl;
+    }
+  } else {
+    // Create default formatter
+    df = DateFormat::createDateTimeInstance(
+        date_style,
+        time_style,
+        displayLocale);
+  }
+
+  if (df == nullptr) {
+    cout << "# DATE TIME FORMATTER == nullptr == " << df << endl;
+    // Post an error in the return
+    return "";
+  }
+
+  // JSON data returned.
+  json_object *return_json = json_object_new_object();
+  json_object_object_add(return_json, "label", label_obj);
+
+  UDate testDateTime;
+  // The type of conversion requested
+
+  // Prefer milliseconds as input.
+  json_object *input_millis = json_object_object_get(json_in, "input_millis");
+  if (input_millis) {
+    testDateTime = json_object_get_double(input_millis);
+  } else {
+    json_object *input_string_obj = json_object_object_get(json_in, "input_string");
+    if (input_string_obj) {
+      const string input_date_string = json_object_get_string(input_string_obj);
+      cout << "# date from input_string: " << input_date_string << endl;
+
+      UnicodeString date_ustring(input_date_string.c_str());
+
+      UnicodeString parse_skeleton = "YYYY-MM-DD HH:mm:ss";
+
+      DateFormat* dparser = DateFormat::createInstanceForSkeleton(parse_skeleton,
+                                                                  displayLocale,
+                                                                  status);
```

Comment:
You shouldn't use an ICU4C DateFormat to parse an ISO-8601 string. You should be able to parse it directly or using a helper library such as `absl`.

Reply:
Good idea! I'll add this.

Reply:
Parsing date string isn't yet working.

Reply:
Fixed this!